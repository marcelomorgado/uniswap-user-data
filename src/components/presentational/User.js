import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import CardActions from "@material-ui/core/CardActions";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";

const styles = theme => ({
  card: {
    minWidth: 275,
  },
  bullet: {
    display: "inline-block",
    margin: "0 2px",
    transform: "scale(0.8)",
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 12,
  },
  button: {
    margin: theme.spacing.unit,
  },
});

function User({ user, style, classes, handleTransactions }) {
  const { id, etherBalance } = user;

  return (
    <Card className={classes.card} style={style}>
      <CardContent>
        <Typography variant="h5" component="h2">
          {id}
        </Typography>
        <Typography className={classes.pos} color="textSecondary">
          {`${etherBalance} ETH`}
        </Typography>
      </CardContent>
      <CardActions>
        <Button
          variant="contained"
          color="primary"
          className={classes.button}
          onClick={() => {
            handleTransactions(id);
          }}
        >
          Transactions
        </Button>
        <Button
          variant="contained"
          color="secondary"
          className={classes.button}
          onClick={() => {}}
        >
          Transfer from
        </Button>
      </CardActions>
    </Card>
  );
}

User.propTypes = {
  user: PropTypes.object.isRequired,
  style: PropTypes.object.isRequired,
  classes: PropTypes.object.isRequired,
  handleTransactions: PropTypes.func.isRequired,
};

//export default User;
export default withStyles(styles)(User);
