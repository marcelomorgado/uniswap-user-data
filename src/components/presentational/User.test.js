import React from "react";
import { shallow } from "enzyme";
import User from "./User";

it("renders the component", () => {
  const user = { id: "0x123User", etherBalance: "10" };
  const props = {
    style: {},
    user,
    handleTransactions: () => {},
  };

  expect(shallow(<User {...props} />)).toMatchSnapshot();
});
