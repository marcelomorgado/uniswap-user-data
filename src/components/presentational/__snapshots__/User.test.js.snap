// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<User
  classes={
    Object {
      "bullet": "User-bullet-2",
      "button": "User-button-5",
      "card": "User-card-1",
      "pos": "User-pos-4",
      "title": "User-title-3",
    }
  }
  handleTransactions={[Function]}
  style={Object {}}
  user={
    Object {
      "etherBalance": "10",
      "id": "0x123User",
    }
  }
/>
`;
